### PR TITLE
Ensure (E)VEX.vvvv bits are set to 1111b when reserved

### LINKED ISSUE #19625: Ensure VEX encoded instrcutions with reserved VEX.vvvv bits have them always set to 1111
Implementation of `AVX2.Permute4x64` #19580 required a change to VEX encoding code that ensured reserved `VEX.vvvv` bits always are set to 1111 otherwise instruction would raise #UD exception.

It turns out that there is large group of VEX and EVEX encoded implemented instructions which require setting `VEX.vvvv` bits to 1111.

Working on a fix.

cc @CarolEidt @fiigii @tannergooding

### PR DESCRIPTION
**Pull request by 4creators**
Fixes #19625

### CODE REVIEW

File: src/jit/emitxarch.cpp [hunk 1]
```diff
@@ -496,6 +496,95 @@ bool emitter::TakesVexPrefix(instruction ins)
...
+// HasReservedVvvvBits: Checks if VEX or EVEX encoding of instruction has reserved VEX.vvvv
+//                      bits which must be set to 1111 otherwise instruction will #UD.
+//
+// Arguments:
+//    instruction -- processor instruction to check
+//
+// Return Value:
+//    true if instruction has a (E)VEX encoding with (E)VEX.vvvv bits reserved
+//
+bool emitter::HasReservedVvvvBits(instruction ins, code_t code)
```
Comment (tannergooding) [inline src/jit/emitxarch.cpp]: I still don't think this is the right fix. This only came up for `vpermpd` and `vpermq`, and was not a problem for the other ~60 instructions.

I think @fiigii's analysis that it was caused by adding `vpermpd` and `vpermq` to the `IsDstDstSrc` method is correct.

Comment (4creators) [reply src/jit/emitxarch.cpp]: I agree that fix for `vpermpd` and `vpermq` is to remove it from `IsDstDstSrc..` check, however, due to the fact that improperly set reserved `VEX.vvvv` bits cause #UD exception we should have such check at the place where those bits could be set.

**Comment by fiigii** [comment]
I do not think we need this fix because `VEX.vvvv` encodes registers in 1's complement, which always encodes the defalut value (XMM0 is 1111b) in `VEX.vvvv` if the instruction is not in DstDstSrc or DstSrcSrc table.

So, the current behavior of RyuJIT is correct.

File: src/jit/emitxarch.cpp (hunk 1 above)
Comment (tannergooding) [reply src/jit/emitxarch.cpp]: I don't think that adding another method to check that is the correct way to do it. These methods (like `IsDstDstSrc`) are already fairly expensive (I saw them taking up to 10% of the emit time, which was on par with `emitInsBinary`).

If an assert is desirable, it should probably be an assert, and should be checked via a flag (which will be possible after I finish the work on https://github.com/dotnet/coreclr/issues/19498).

Comment (4creators) [reply src/jit/emitxarch.cpp]: Agree with the arguments and that assert is enough to get it double checked.

**Pull request closed without merging**